src/sql/workbench/contrib/notebook/find/notebookFindModel.ts

```diff
@@ -585,14 +584,50 @@ export class NotebookFindModel extends Disposable implements INotebookFindModel
 	// In markdown links are defined as [Link Text](https://url/of/the/text). when searching for text we shouldn't
 	// look for the values inside the (), below regex replaces that with just the Link Text.
 	cleanMarkdownLinks(cellSrc: string): string {
-		return cellSrc.replace(/(?:__|[*#])|\[(.*?)\]\(.*?\)/gm, '$1');
+		return cellSrc.replace(/(?:__)|\[([\s\S]*?)\]\(.*?\)/gm, '$1');
 	}
 
 	// remove /n's to calculate the line number to locate the correct element
 	cleanUpCellSource(cellValue: string | string[]): string | string[] {
 		let trimmedCellSrc: string[] = [];
+		const newline = new RegExp(/^([-=]+[\\r\\n]*)$/gm); // new RegExp(/^([\s-=]+[\\r\\n]*)$/gm);
+		const tableSeperator = new RegExp(/^(([|]+[ :]*[-]+[ :]*[|]*)+[\\r\\n]*)$/gm);
+		const tableRows = new RegExp(/^\|(.+\|*)*/gm);
+		const listRegex = new RegExp(/^[*+-]+(.+[\r\n]*)/gm);// ^[*+-]+[\s]*.+
+		const blockQuoutesRgx = new RegExp(/^\>(.*)/gm);
+		const headingsRegex = new RegExp(/(^\#+(.*))|(.*?\n+[-=]+)$/gm); //new RegExp(/^\#+(.*)/gm);
 		if (cellValue instanceof Array) {
-			trimmedCellSrc = cellValue.filter(c => c !== '\n' && c !== '\r\n' && c.indexOf('|-') === -1);
+			let paragraphContent: string = '';
+			cellValue.forEach(content => {
+				if (content !== '\n' && content !== '\r\n' && !content.match(tableSeperator) && !content.match(newline) && !content.match(listRegex) && !content.match(blockQuoutesRgx) && !content.match(tableRows) && !content.match(headingsRegex)) {
```

Comment:
There's a lot going on here - please add comments explaining what is supposed to be going on at each check. (helps both reviewing it and for future people looking at this to understand what this is all about)